### PR TITLE
Add strategy flag to deploy

### PR DESCRIPTION
**Pull request by radito3**
#### Description: 
Add --strategy flag to deploy command

### CODE REVIEW

File: commands/blue_green_deploy_command.go [hunk 1]
```diff
@@ -8,7 +8,10 @@ import (
 	"github.com/cloudfoundry/cli/plugin"
 )
 
-const noConfirmOpt = "no-confirm"
+const (
+	noConfirmOpt = "no-confirm"
+	keepExistingNamesOpt = "keep-existing-names"
```
Comment (enchobelezirev) [inline commands/blue_green_deploy_command.go]: From the name of the option there is no clear way of understanding which names would be kept(names of applications or services). Maybe the option should be extended in order to contain more information about which names will be kept.

File: commands/base_command.go [hunk 2]
```diff
@@ -29,7 +29,6 @@ import (
 
 const (
 	// DeployServiceURLEnv is the deploy service URL environment variable
-	DeployServiceURLEnv           = "DEPLOY_SERVICE_URL"
```
Comment (IvanBorislavovDimitrov) [inline commands/base_command.go]: Why is this removed?

File: commands/base_command.go [hunk 3]
```diff
@@ -39,7 +38,6 @@ const (
 	noRestartSubscribedAppsOpt    = "no-restart-subscribed-apps"
 	noFailOnMissingPermissionsOpt = "do-not-fail-on-missing-permissions"
 	abortOnErrorOpt               = "abort-on-error"
-	deployServiceHost             = "deploy-service"
```
Comment (IvanBorislavovDimitrov) [inline commands/base_command.go]: Why is this removed?

File: commands/deploy_command.go [hunk 4]
```diff
@@ -317,6 +316,14 @@ func (c *DeployCommand) Execute(args []string) ExecutionStatus {
 	processBuilder.Parameter("mtaId", mtaID)
 	setModulesAndResourcesListParameters(modulesList, resourcesList, processBuilder, mtaElementsCalculator)
 	c.processParametersSetter(optionValues, processBuilder)
+
+	strategy := GetStringOpt(strategyOpt, optionValues)
+	if len(strategy) > 0 && strategy == "blue-green" {
```
Comment (IvanBorislavovDimitrov) [inline commands/deploy_command.go]: Do you need len(strategy) > 0. In Go empty/nil strings are always "".

File: commands/file_uploader.go [hunk 5]
```diff
@@ -62,7 +62,9 @@ func (f *FileUploader) UploadFiles() ([]*models.FileMetadata, ExecutionStatus) {
 		if !isFileAlreadyUploaded(file, fileInfo, uploadedMtaFiles, &alreadyUploadedFiles) {
 			// If not, add it to the list of uploaded files
 			fileToUpload, err := os.Open(file)
-			defer fileToUpload.Close()
+			if fileToUpload != nil {
```
Comment (IvanBorislavovDimitrov) [inline commands/file_uploader.go]: This nil check is redundant because fileToUpload will be nil only if an error occurs. Even if an error occurs because of the defer statement Close() will be executed after the surrounding function returns.

File: commands/file_uploader_test.go [hunk 6]
```diff
@@ -47,7 +47,7 @@ var _ = Describe("FileUploader", func() {
 					uploadedFiles, status = fileUploader.UploadFiles()
 				})
 				ex.ExpectSuccess(status.ToInt(), output)
-				Expect(uploadedFiles).To(Equal([]*models.FileMetadata{}))
+				Expect(uploadedFiles).To(BeNil())
```
Comment (IvanBorislavovDimitrov) [inline commands/file_uploader_test.go]: Why are these expected to be nil?

File: clients/csrf/fakes/fake_csrf_token_fetcher.go [hunk 7]
```diff
@@ -11,7 +11,7 @@ const FakeCsrfTokenValue = "fake-xcsrf-token-value"
 type FakeCsrfTokenFetcher struct {
 }
 
-func (c *FakeCsrfTokenFetcher) FetchCsrfToken(url string, currentRequest *http.Request) (*csrf_paramters.CsrfRequestHeader, error) {
+func (c *FakeCsrfTokenFetcher) FetchCsrfToken(string, *http.Request) (*csrf_paramters.CsrfRequestHeader, error) {
```
Comment (IvanBorislavovDimitrov) [inline clients/csrf/fakes/fake_csrf_token_fetcher.go]: Are you sure you still implement CsrfTokenFetcher?

File: commands/base_command.go (hunk 2 above)
Comment (radito3) [reply commands/base_command.go]: It wasn't used anywhere

File: commands/base_command.go (hunk 3 above)
Comment (radito3) [reply commands/base_command.go]: It wasn't used anywhere

File: commands/file_uploader.go (hunk 5 above)
Comment (VaskoBozhurski) [reply commands/file_uploader.go]: You can just move the defer after the if err != nil check on line 68.  os.Open would either return an error or the file handle, but not both.

File: commands/file_uploader_test.go (hunk 6 above)
Comment (radito3) [reply commands/file_uploader_test.go]: Because when I changed it from 
`filesToUpload := []os.File{}`
to 
`var filesToUpload []os.File` 
the default value is nil, not an empty array

File: commands/file_uploader.go (hunk 5 above)
Comment (radito3) [reply commands/file_uploader.go]: Done

File: commands/deploy_command.go (hunk 4 above)
Comment (radito3) [reply commands/deploy_command.go]: Changed to `if strategy == "blue-green"`
